libsolidity/codegen/Compiler.h

```diff
@@ -59,12 +59,12 @@ class Compiler
 	/// @returns Only the runtime object (without constructor).
 	eth::LinkerObject runtimeObject() const { return m_context.assembledRuntimeObject(m_runtimeSub); }
 	/// @arg _sourceCodes is the map of input files to source code strings
-	std::string assemblyString(StringMap const& _sourceCodes = StringMap()) const
+	std::string assemblyString(StringMap _sourceCodes = StringMap()) const
 	{
 		return m_context.assemblyString(_sourceCodes);
```

Comment:
In these cases you need to ``std::move`` the argument to pass it on by moving instead of copying.